### ISSUE TITLE
Improve electron-store performance

### ISSUE THREAD

**Issue by fxha** [marktext/marktext#1108]
### Description

We should fork `electron-store` (`conf`) and only read the configuration from disk if the file has changed but not on every `get` call. At the same time we could remove unnecessary stuff like encryption.

**Comment by Jocs** [comment]
Fork a the codes and then maintain it ourselves. There are also modifications and maintenance costs. You may need to consider the trade-off between benefits and costs, or submit an issue directly to electron-store?

**Comment by fxha** [comment]
Yes, there is an opened issue from early 2018 without process nor comments. That's true, we could pull upstream changes and use `chokidar` to watch and read changes into memory. I don't know the exact use case of `electron-store` but I think this would be the best way for us.

**Comment by Jocs** [comment]
Thank you for this suggestion. The performance concern you identified is still technically present — `conf` (the library underlying `electron-store`) does read from disk on every store access rather than caching in memory. The team is already aware of this; there is an internal TODO in `preferences/index.ts` tracking it.

That said, in practice the impact on a markdown editor is minimal: user preferences are loaded once per window at startup, not in any tight rendering loop. After 7 years without a reported user-visible slowdown and given the maintenance cost of forking a third-party library, we are closing this issue as a low-priority item. The in-code TODO will keep this on the radar if we revisit the configuration layer in the future. If you are experiencing concrete, measurable lag related to preference reads, please open a new issue with a profiling trace and we are happy to revisit.
